lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +26,33 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
+        servers = {}
+
         response = compute_client.list_instances(compartment_id, options)
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        if config[:format] == 'summary' && response.data
+          # build a hash of hashs, keyed off of server_id
+          vnics = compute_client.list_vnic_attachments(compartment_id, {})
+          vnics.data.each do |vnic|
+            servers[vnic.instance_id] = {}
+            servers[vnic.instance_id]['vnic_id'] = vnic.vnic_id
+          end
+
+          # add the private_ip/public_ip addresses to the hash
+          servers.each do |server, vnic_info|
```

Comment:
A bit of a nitpick, but it would be simpler to just get the ip addresses in the first loop.

You'll also need to look out for 404's when calling get_vnic, in the case of terminated instances.